GeneratorInterface/GenFilters/plugins/LHEPtFilter.cc

```diff
@@ -0,0 +1,121 @@
+// -*- C++ -*-
+//
+// Package:    LHEPtFilter
+// Class:      LHEPtFilter
+//
+/* 
+
+ Description: Filter to select events with pT in a given range.
+ (Based on LHEGenericFilter)
+
+     
+*/
+//
+
+// system include files
+#include <memory>
+#include <iostream>
+#include <set>
+
+// user include files
+#include "Math/Vector4D.h"
+#include "Math/Vector4Dfwd.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDFilter.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "SimDataFormats/GeneratorProducts/interface/LHEEventProduct.h"
+
+//
+// class declaration
+//
+
+class LHEPtFilter : public edm::EDFilter {
```

Comment:
Could this be made a non-legacy module, preferably global or stream?

Reply:
Yes, has been converted to a global module in the latest commits.